src/Compilers/CSharp/Test/Symbol/SymbolDisplay/SymbolDisplayTests.cs

```diff
@@ -7652,5 +7652,185 @@ public void RecordDeclaration()
+                SymbolDisplayPartKind.PropertyName);
+            Verify(
+                openGenericMemmber.ToDisplayParts(format),
+                "T? P",
+                SymbolDisplayPartKind.TypeParameterName,
+                SymbolDisplayPartKind.Punctuation,
+                SymbolDisplayPartKind.Space,
+                SymbolDisplayPartKind.PropertyName);
+
+            format = format.WithMiscellaneousOptions(SymbolDisplayMiscellaneousOptions.None); // without SymbolDisplayMiscellaneousOptions.IncludeNullableReferenceTypeModifier
+            Verify(
+                closedGenericMemmber.ToDisplayParts(format),
+                "Object P",
+                SymbolDisplayPartKind.ClassName,
+                SymbolDisplayPartKind.Space,
+                SymbolDisplayPartKind.PropertyName);
+            Verify(
+                openGenericMemmber.ToDisplayParts(format),
+                "T P",
+                SymbolDisplayPartKind.TypeParameterName,
+                SymbolDisplayPartKind.Space,
+                SymbolDisplayPartKind.PropertyName);
+        }
+
+        [Fact]
+        [WorkItem(48023, "https://github.com/dotnet/roslyn/issues/48023")]
+        public void NullableAttributes_MayBeNullOnStructProperty()
+        {
+            var source = @"
+using System.Diagnostics.CodeAnalysis;
+
+#nullable enable
+
+public class A<T>
+{
+    [MaybeNull]
+    public T P { get; }
+}
+
+public class Test: A<int> { }
+";
+            var compilation = CreateCompilation(new[] { source, MaybeNullAttributeDefinition });
+            var closedGenericMember = compilation.GetTypeByMetadataName("Test").BaseType().GetMember("P");
+            var format = new SymbolDisplayFormat(
+                memberOptions: SymbolDisplayMemberOptions.IncludeParameters | SymbolDisplayMemberOptions.IncludeType | SymbolDisplayMemberOptions.IncludeModifiers,
+                genericsOptions: SymbolDisplayGenericsOptions.IncludeTypeParameters,
+                miscellaneousOptions: SymbolDisplayMiscellaneousOptions.IncludeNullableReferenceTypeModifier);
+
+            Verify(
+                closedGenericMember.ToDisplayParts(format),
+                "Int32 P",
+                SymbolDisplayPartKind.StructName,
+                SymbolDisplayPartKind.Space,
+                SymbolDisplayPartKind.PropertyName);
+        }
+
+        [WorkItem(48023, "https://github.com/dotnet/roslyn/issues/48023")]
+        [Fact]
+        public void NullableAttributes_VisualBasicReferencesCSharp()
+        {
+            var sourceCS = @"
+using System.Diagnostics.CodeAnalysis;
+
+#nullable enable
+
+namespace N
+{
+    public class A<T>
+    {
+        [MaybeNull]
+        public T P { get; }
+    }
+}
+";
+            var sourceVB = @"
+Imports N
+Public Class Test
+    Inherits A(Of Object)
+End Class";
+            var compCS = CreateCompilation(new[] { sourceCS, MaybeNullAttributeDefinition });
+            var refCS = compCS.EmitToImageReference();
```

Comment:
There is also `ToMetadataReference()`. I'm not sure I picked the right method here, or if it makes sense to test with both kinds of references (the same is true for the next test `NullableAttributes_CSharpReferencesVisualBasic`)

Reply:
>There is also ToMetadataReference(). I'm not sure I picked the right method here, or if it makes sense to test with both kinds of references (the same is true for the next test NullableAttributes_CSharpReferencesVisualBasic)

That one returns a compilation reference. Compilation references cannot be used across language "boundary".

---
In reply to: [559173858](https://github.com/dotnet/roslyn/pull/50527#discussion_r559173858) [](ancestors = 559173858)